src/viam/examples/modules/complex/main.cpp

```diff
@@ -0,0 +1,98 @@
+#include <memory>
+#include <signal.h>
+
+#include <boost/log/trivial.hpp>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/server_context.h>
+
+#include <viam/api/common/v1/common.grpc.pb.h>
+#include <viam/api/component/generic/v1/generic.grpc.pb.h>
+#include <viam/api/robot/v1/robot.pb.h>
+
+#include <viam/sdk/components/base/base.hpp>
+#include <viam/sdk/components/component.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/module/module.hpp>
+#include <viam/sdk/module/service.hpp>
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource.hpp>
+#include <viam/sdk/rpc/dial.hpp>
+#include <viam/sdk/rpc/server.hpp>
+
+#include "base/impl.hpp"
+#include "gizmo/api.hpp"
+#include "gizmo/impl.hpp"
+#include "summation/api.hpp"
+#include "summation/impl.hpp"
+
+using namespace viam::sdk;
+
+int main(int argc, char** argv) {
+    if (argc < 2) {
+        throw std::invalid_argument("Need socket path as command line argument");
```

Comment:
per your comment elsewhere, should we return `EXIT_FAILURE` rather than throwing?